torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -500,6 +576,10 @@ bool FunctionSignature::parse(PyObject* args, PyObject* kwargs, PyObject* dst[],
     return false;
   }
 
+  if (!overloaded_args.empty()) {
+    overloaded_args.clear();
+  }
```

Comment:
This is necessary because now that I've moved  `overloaded_args` to live on `FunctionSignature`, RAII is no longer deleting `overloaded_args` after each function call. It might be clearer if I put this inside of `handle_torch_function` when I'm done with `overloaded_args` there.